### ISSUE TITLE
Unify rendering for ListBox

### ISSUE THREAD

**Issue by silviyaboteva** [telerik/kendo-themes#3318]
## Overview

Determine the common rendering for the ListBox component that can be used across the different suites. 

Make sure the proposed rendering takes into consideration the guidelines from the accessibility team and [the related component spec](https://github.com/telerik/web-components-ux/tree/master/docs/components).

### Proposed Rendering

<details>

<summary>Universal</summary>

```html
<!-- default listbox rendering -->
<div class="k-listbox k-listbox-md k-listbox-actions-left">
    <div class="k-listbox-actions">
        <button class="k-button k-button-md k-button-solid k-button-solid-base k-rounded-md k-icon-button">
            <span class="k-button-icon k-icon k-i-#{$iconName}">...</span>
        </button>
        <button class="k-button k-button-md k-button-solid k-button-solid-base k-rounded-md k-icon-button">
            <span class="k-button-icon k-icon k-i-#{$iconName}">...</span>
        </button>
        ....
    </div>
    <div class="k-list-scroller k-selectable">
        <div class="k-list k-list-md">
            <div class="k-list-content">
                <ul class="k-list-ul">
                    <li class="k-list-item">
                        <span class="k-list-item-text">...</span>
                    </li>
                    <li class="k-list-item">
                        <span class="k-list-item-text">...</span>
                    </li>
                    ...
                </ul>
            </div>
        </div>
    </div>
</div>
```

</details>

### Changes
1. `k-widget` class removed
1. `k-listbox-toolbar` class replaced with `k-listbox-actions`
1. `k-listbox-toolbar-left`,  `k-listbox-toolbar-right`, `k-listbox-toolbar-top`, `k-listbox-toolbar-bottom` classes replaced with `k-listbox-actions-...`
1. listbox actions which in some suites were represented by `<a>` have been replaced with `<button/>`
1. listbox actions are now wrapped inside a single container
```html
    <div class="k-listbox-actions">
        <button class="k-button k-button-md k-button-solid k-button-solid-base k-rounded-md k-icon-button">
            <span class="k-button-icon k-icon k-i-#{$iconName}">...</span>
        </button>
    </div>
```
### Visual Tests

https://github.com/telerik/kendo-themes/tree/listbox-unified-rendering/tests/listbox

## Deliverables

- [x] Universal rendering
- [x] HTML Component
- [x] HTML JSON Spec
- [x] Rename variables to `$kendo-[varName]`
- [x] Update tests

## Verification

Tag the `telerik/kendo-a11y` group for review of the rendering.

**Comment by jivanova** [comment]
The accessibility team (@veselints) was working on a specification for the ListBox component. We discussed that to achieve better a11y and assure no violation we might need to change the rendering of the component.

**Accessibility points**

- ListBox toolbar items should be focusable and `aria-controls` should point to the listboxes that are controlled by the toolbar
- Toolbar should be rendered in semantic html sequence and should not have positioning in the themes 

<details>
<summary>WCAG Techniques</summary>

https://www.w3.org/TR/WCAG20-TECHS/css 

The objective of this technique is to demonstrate how visual appearance may be enhanced via style sheets while still maintaining a meaningful presentation when style sheets are not applied. Using the positioning properties of CSS2, content may be displayed at any position on the user's viewport. Using structural elements ensures that the meaning of the content can still be determined when styling is not available.

</details>
